### PR TITLE
PRE check

### PR DESCRIPTION
**Pull request by michalsmiarowski**
Change button in Authorization Table based on the status of PRE contract for given stake.

Main changes:
- add SimplePREApplication contract (for now we have to paste the address by hand when setting the project locally until we figure out some better solution)
- restyle button based on the status of PRE contract for given stake
  <details>
  <summary>example</summary>
  <br> 
  PRE node not set up:<br />
  image<br /><br />
   PRE node set up:<br />
  image
 </details>
 
- add links for tooltips in Threshold Authorization table (see example above)

### CODE REVIEW

File: solidity-v1/dashboard/src/lib/keep/index.js [hunk 1]
```diff
@@ -199,10 +207,26 @@ class Keep {
     this.keepToTStaking = new KeepToTStaking(
       this.thresholdStakingContract,
       this.thresholdKeepStakeContract,
+      this.simplePREApplicationContract,
       this.web3
     )
   }
 
+  _initializePREContract = async (web3) => {
+    const chainId = await web3.lib.eth.getChainId()
```
Comment (r-czajkowski) [inline solidity-v1/dashboard/src/lib/keep/index.js:216]: We can get chain id using `getChainId` fn in `src/connectors/utils`. Or let's update the keep lib a bit and add chain id to the `Keep` constructor so we make the lib independent of dapp implementation.

File: solidity-v1/dashboard/src/lib/keep/index.js [hunk 2]
```diff
@@ -199,10 +207,26 @@ class Keep {
     this.keepToTStaking = new KeepToTStaking(
       this.thresholdStakingContract,
       this.thresholdKeepStakeContract,
+      this.simplePREApplicationContract,
       this.web3
     )
   }
 
+  _initializePREContract = async (web3) => {
```
Comment (r-czajkowski) [inline solidity-v1/dashboard/src/lib/keep/index.js:215]: Let's call this function inside `initializeContracts`.

File: solidity-v1/dashboard/src/lib/keep/index.js [hunk 3]
```diff
@@ -41,7 +44,9 @@ class Keep {
   static initialize(web3) {
     const keep = new Keep(web3)
     keep.initializeContracts()
-    keep.initializeServices()
+    keep._initializePREContract(web3).then(() => {
+      keep.initializeServices()
+    })
```
Comment (r-czajkowski) [inline solidity-v1/dashboard/src/lib/keep/index.js:49]: If you agree with the above comments we can leave as it was:
```
keep.initializeContracts()
keep.initializeServices()
```

File: solidity-v1/dashboard/package-lock.json [hunk 4]
```diff
@@ -3216,7 +3216,7 @@
     "@stablelib/binary": {
       "version": "0.7.2",
       "resolved": "https://registry.npmjs.org/@stablelib/binary/-/binary-0.7.2.tgz",
-      "integrity": "sha1-GzOSFwyKh0HIuPhD6ilN5xrrLPc=",
+      "integrity": "sha512-J7iGppeKR112ICTZTAoALcT3yBpTrd2Z/F0wwiOUZPVPTDFTQFWHZZdYzfal9+mY1uMUPRSEnNmDuXRZbtE8Xg==",
```
Comment (r-czajkowski) [inline solidity-v1/dashboard/package-lock.json:3219]: I can't see any changes in `package.json` file so I think `pacakge-lock.json` should not be updated 🤔.

**Review by r-czajkowski — CHANGES_REQUESTED** [review]

File: solidity-v1/dashboard/package-lock.json (hunk 4 above)
Comment (michalsmiarowski) [reply solidity-v1/dashboard/package-lock.json]: 19ecb12dc0b9e1ecf836865841199a7a739b50ac

File: solidity-v1/dashboard/src/lib/keep/index.js (hunk 1 above)
Comment (michalsmiarowski) [reply solidity-v1/dashboard/src/lib/keep/index.js]: a3ab04267c9d6acc69725ef3e469c8817810ac8e

File: solidity-v1/dashboard/src/lib/keep/index.js (hunk 2 above)
Comment (michalsmiarowski) [reply solidity-v1/dashboard/src/lib/keep/index.js]: a3ab04267c9d6acc69725ef3e469c8817810ac8e

File: solidity-v1/dashboard/src/lib/keep/index.js (hunk 3 above)
Comment (michalsmiarowski) [reply solidity-v1/dashboard/src/lib/keep/index.js]: a3ab04267c9d6acc69725ef3e469c8817810ac8e

File: solidity-v1/dashboard/src/lib/keep/index.js (hunk 1 above)
Comment (michalsmiarowski) [reply solidity-v1/dashboard/src/lib/keep/index.js]: Changed a little bit in 42f1b04dc22da70f4cd4b1ac507181aeb94ae941

**Review by r-czajkowski — APPROVED** [review]
:shipit:

**Pull request merged**